### PR TITLE
ci: fix TF attestation version assertion in integration test

### PR DESCRIPTION
**Pull request by elchead**
### Context

The attestation version numbers were hard-coded in the test but they are dynamically updated.

### Proposed change(s)

- soften version value assertion to be non-zero where applicable

### Additional info

- [AB#4381](https://dev.azure.com/Edgeless/ae37573d-ccde-4af2-ab1e-001e587197d1/_workitems/edit/4381)

### Checklist

- [x] Run the E2E tests that are relevant to this PR's changes
- [x] Update [docs](https://github.com/edgelesssys/constellation/tree/main/docs)
- [x] Add labels (e.g., for changelog category)
- [x] Is PR title adequate for changelog?
- [x] Link to Milestone

### CODE REVIEW

File: terraform-provider-constellation/internal/provider/attestation_data_source_test.go [hunk 1]
```diff
@@ -16,6 +17,21 @@ func TestAccAttestationSource(t *testing.T) {
 	// Set the path to the Terraform binary for acceptance testing when running under Bazel.
 	bazelPreCheck := func() { bazelSetTerraformBinaryPath(t) }
 
+	assertNonZeroValue := func(attr string) resource.TestCheckFunc {
+		return resource.TestMatchResourceAttr(
+			"data.constellation_attestation.test",
+			attr,
+			regexp.MustCompile(`^[1-9]\d*$`),
```
Comment (burgerdev) [inline terraform-provider-constellation/internal/provider/attestation_data_source_test.go:24]: Could we rather use https://pkg.go.dev/github.com/hashicorp/terraform-plugin-testing/helper/resource#TestCheckResourceAttrWith? I fear that regexp matching of numbers is somewhat error prone.

Comment (elchead) [reply terraform-provider-constellation/internal/provider/attestation_data_source_test.go:24]: I guess everyone hates regex :D it matches only digits. How is this error prone?
But I can also rewrite the logic in Go.

Comment (burgerdev) [reply terraform-provider-constellation/internal/provider/attestation_data_source_test.go:24]: This one does not match `012`, which may or may not be a valid uint8, depending on who you ask. The other one matches `foo1bar`, which is too permissive I believe.

I'm not against regexps per se, but https://regex.info/blog/2006-09-15/247.

**Review by burgerdev — APPROVED** [review]
Great, thanks!

**Pull request merged**